### DISCUSSION TITLE
Have next/image use the size of the image.

### DISCUSSION THREAD

**Discussion by Arcath** [vercel/next.js#21633] [Help]
I have some markdown content with images in. I'm providing some custom components to improve the rendering of the posts and I'm hitting an issue with my custom `img` using `next/image`.

I'm not sure what the best way to do this is. I want my images to be what ever size the image happens to be upto a maximum width of the screen width.

`layout="fill"` is the only way to not specify the image containers size to the pixel but that results in zero height divs and simple moves the issue to the parent container.

Am I missing something? Is there a way to make `next/image` act positionally like `img`?

**Comment by NickBeukema** [comment]
I have a similar use-case as well. We're using a headless CMS to insert images, and historically we've approached these images with the "windowed" mode, or "cover" display, given a width/height it will fill the space but keep proportions.

Similar to @Arcath I'm seeing the only way to receive the image without specially giving height/width info (because we don't know it), is to use the `layout="fill"` property.

Could I suggest something similar to how Contentful approaches this: They allow passing _either_ a height or a width, and they proportionally scale the image. That way we can say "I never want my image to be displayed wider than X pixels" and the height will be calculated based on that, or vice-versa for height.

**Comment by raffij** [comment]
To optimize content layout shift you'll always need to artificially set the aspect ratio of the image or use those of the original image. We store the original aspect ratio of the uploaded image and pass that to next/image component in responsive mode.

**Reply by NickBeukema** [reply]
Not necessarily. For the case I'm referring to we have a window, say 200x200, but the image could be 400x600. In this case you would shrink the image to 200x300 and display a portion of it (cover in css). In this case, you can appropriately assume for content shifting that the box will be 200x200, but this is not necessarily tied to the aspect ratio of the image.

**Comment by Glutch** [comment]
I have the same problem, this seems so strange. I upload images to s3 bucket, do i really need to save the image height & width and later add that to the Image component while manually calculating aspect ratio to get a normal width: '100%'?

**Comment by michael-goss** [comment]
I ran into a similar problem where I wanted to fill my next/image the entire width of its parent element dynamically without taking care of aspect ratios or widths and heights. I managed to solve it with some CSS:
Apparently the ```<Image />``` element translates into ```<div><img/></div>``` with absolute positioning.

Wrap your ```<Image />``` like this:
```html
<div className={styles.imageWrapper}>
    <Image ... layout="fill" objectFit="contain"/>
</div>
```
with 
```css
.imageWrapper {
    width: 100%; /* image shall have 100% width of parent, use max-width alternatively */
}

.imageWrapper > div {
    position: static !important;
}

.imageWrapper > div > img {
   position: static !important;
   width: 100% !important;
   height: 100% !important;
}
```

Not a fan of using ```!important``` rule though. Unfortunately, doesn't work otherwise.
Lmk if you find a better way

**Reply by Glutch** [reply]
It works by manually assigning the styles, but this doesn't work. imageWrapper exists on the parent, but the styles of imageWrapper > div and for the img are missing. Very strange, any ideas?

```javascript
<div className="imageWrapper">
  <Image src="/material/wheels4u.jpg" layout="fill" objectFit="contain" />
</div>
```
```jsx
<style jsx>{`
  .imageWrapper {
    width: 100%;
  }

  .imageWrapper > div {
    position: static !important;
  }

  .imageWrapper > div > img {
    position: static !important;
    width: 100% !important;
    height: 100% !important;
  }
`}</style>
 ```

**Reply by Glutch** [reply]
nvm, it worked by putting the css in globals.css instead of using <style jsx>. Thank you for the solution!

**Reply by angiehemans** [reply]
Thank you so much @michael-goss this is exactly the fix I was looking for!

**Reply by RowinVanAmsterdam** [reply]
Somehow my img was wrapped in a span-tag, instead of a div. So I had to make a small adjustment there and perhaps worth checking if the solution mentioned by @Glutch is not working. 


```
.imageWrapper > span

.imageWrapper > span > img
```
